[upnpcontrol] Add media feature to simplify handling

lo92fr opened the pull request:
# Add a new feature media to simplify handling of media in Openhab

# Description

This pull request replace previous #18501 pull request (divide modification on differents binding into separate branches).

See this issue for description :
https://github.com/openhab/openhab-core/issues/4710

Depends on: https://github.com/openhab/openhab-core/pull/4709

lo92fr committed:
feature-media-spotify : divide branch feature-media to separate branches

Signed-off-by: Laurent ARNAL <laurent@clae.net>

lo92fr committed:
spotless:apply

Signed-off-by: Laurent ARNAL <laurent@clae.net>

lo92fr committed:
fix upnp handling for new MediaType impl

Signed-off-by: Laurent ARNAL <laurent@clae.net>

mherwege commented:
Please make sure the name of the PR contains the binding name. Multiple PR’s with the same name are hard to distinguish.

lo92fr commented:
> Please make sure the name of the PR contains the binding name. Multiple PR’s with the same name are hard to distinguish.

Hello Mark,

I've fixed them.
Sorry for it.

Laurent.

andrewfg commented:
I like this concept.

lo92fr committed:
Starting adding device support to Upnp binding

Signed-off-by: Laurent ARNAL <laurent@clae.net>

lo92fr committed:
progress on support feature-media on upnp controller

Signed-off-by: Laurent ARNAL <laurent@clae.net>

lo92fr committed:
fix compilation errors

Signed-off-by: Laurent ARNAL <laurent@clae.net>

lo92fr committed:
implements paging and lazy loading on MediaCollection (for now only test / implements on upnp Binding)

Signed-off-by: Laurent ARNAL <laurent@clae.net>

lo92fr committed:
add support for global media player on upnp binding

Signed-off-by: Laurent ARNAL <laurent@clae.net>

lo92fr committed:
review search functionnalites (WIP)

Signed-off-by: Laurent ARNAL <laurent@clae.net>

lo92fr committed:
code cleanup : introduce MediaCommandType & MediaStateType

Signed-off-by: Laurent ARNAL <laurent@clae.net>

lo92fr committed:
fix compile : need to review review rebase step to be sure don't miss anything

Signed-off-by: Laurent ARNAL <laurent@clae.net>

lo92fr committed:
fix errors in last commits

Signed-off-by: Laurent ARNAL <laurent@clae.net>

lo92fr committed:
introduce playback from Tidal, work but very dirty, need review

Signed-off-by: Laurent ARNAL <laurent@clae.net>

lo92fr committed:
add missing TrackMetaData key in onValueReceived
clean up media rendering from other binding using getStreamUri from mediaService

Signed-off-by: Laurent ARNAL <laurent@clae.net>

lo92fr committed:
add TrackMetadata keys

Signed-off-by: Laurent ARNAL <laurent@clae.net>

lo92fr committed:
add primary support for Media Queue

Signed-off-by: Laurent ARNAL <laurent@clae.net>

lo92fr committed:
code refactoring to remove duplicate code between addon that use mediaService and simplify collection handling

Signed-off-by: Laurent ARNAL <laurent@clae.net>

lo92fr committed:
progress on tidal playback, working version with direct mpd play from mpd Stream

Signed-off-by: Laurent ARNAL <laurent@clae.net>

lo92fr committed:
review naming :
MediaDevice / MediaSink
MediaSource / MediaCollectionSource

Signed-off-by: Laurent ARNAL <laurent@clae.net>

lo92fr committed:
review sink registration

Signed-off-by: Laurent ARNAL <laurent@clae.net>

lo92fr committed:
experiment playing stream to Java audio or WebAudio sink

Signed-off-by: Laurent ARNAL <laurent@clae.net>

lo92fr committed:
move to 5.2.0-SNAPSHOT

Signed-off-by: Laurent ARNAL <laurent@clae.net>
